src/analysis/validation.jl

```diff
@@ -2,102 +2,238 @@
+        head_loss_wntr = vec(head_loss_wntr), head_loss_watermodels = vec(head_loss_watermodels) )
+
+    return pipe_df
+end
+
+# compare short pipes
+function get_short_pipe_dataframe(wm_data,wm_solution,wntr_data,wntr_simulation,short_pipe_id)
+    duration = length(wm_solution["solution"]["nw"])            # number of time steps
+    time_step = wm_data["time_step"]/3600                       # length per time step (hour)
+    short_pipe_name = wm_data["nw"]["1"]["short_pipe"][short_pipe_id]["source_id"][2]
+    flow_wntr = Array{Float64,1}(undef,duration)
+    flow_watermodels = Array{Float64,1}(undef,duration)
+    head_loss_wntr = Array{Float64,1}(undef,duration)
+    head_loss_watermodels = Array{Float64,1}(undef,duration)
+
+    for t in 1:duration
+        flow_wntr[t] = wntr_simulation.link["flowrate"][short_pipe_name].values[t]
+        flow_watermodels[t] = wm_solution["solution"]["nw"][string(t)]["short_pipe"][short_pipe_id]["q"]
+        start_node = wntr_data.links._data[short_pipe_name].start_node_name
+        end_node = wntr_data.links._data[short_pipe_name].end_node_name
+        head_loss_wntr[t] = wntr_simulation.node["head"][start_node].values[t]-wntr_simulation.node["head"][end_node].values[t]
+        head_loss_watermodels[t] = -1*wm_solution["solution"]["nw"][string(t)]["short_pipe"][short_pipe_id]["dhn"]+wm_solution["solution"]["nw"][string(t)]["short_pipe"][short_pipe_id]["dhp"]
+    end
+    
+    short_pipe_df = DataFrame(time = 1:time_step:time_step*duration, flow_wntr = vec(flow_wntr), flow_watermodels = vec(flow_watermodels), 
+        head_loss_wntr = vec(head_loss_wntr), head_loss_watermodels = vec(head_loss_watermodels) )
+
+    return short_pipe_df
+end
+
+# compare valves
+function get_valve_dataframe(wm_data,wm_solution,wntr_data,wntr_simulation,valve_id)
+    duration = length(wm_solution["solution"]["nw"])            # number of time steps
+    time_step = wm_data["time_step"]/3600                       # length per time step (hour)
+    valve_name = wm_data["nw"]["1"]["valve"][valve_id]["source_id"][2]
+    flow_wntr = Array{Float64,1}(undef,duration)
+    flow_watermodels = Array{Float64,1}(undef,duration)
+    head_loss_wntr = Array{Float64,1}(undef,duration)
+    head_loss_watermodels = Array{Float64,1}(undef,duration)
+
+    for t in 1:duration
+        flow_wntr[t] = wntr_simulation.link["flowrate"][valve_name].values[t]
+        flow_watermodels[t] = wm_solution["solution"]["nw"][string(t)]["valve"][valve_id]["q"]
+        start_node = wntr_data.links._data[valve_name].start_node_name
+        end_node = wntr_data.links._data[valve_name].end_node_name
+        head_loss_wntr[t] = wntr_simulation.node["head"][start_node].values[t]-wntr_simulation.node["head"][end_node].values[t]
+        head_loss_watermodels[t] = -1*wm_solution["solution"]["nw"][string(t)]["valve"][valve_id]["dhn"]+wm_solution["solution"]["nw"][string(t)]["valve"][valve_id]["dhp"]
```

Comment:
Same comment as above when computing head differences for the WaterModels solution.